kernel/src/utilities/conversions.rs

```diff
@@ -0,0 +1,37 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2025.
+
+//! Helper functions for converting values and data types.
+
+/// Helper function to convert create a full usize value from two 32-bit usize
+/// values.
+///
+/// In C this would look like:
+///
+/// ```c
+/// size_t v = (hi << 32) | (uint32_t) lo;
+/// ```
+///
+/// This is useful when passing a machine-sized value (i.e. a `size_t`) via the
+/// system call interface in two 32-bit usize values. On a 32-bit machine this
+/// essentially has no effect; the full value is stored in the `lo` usize. On a
+/// 64-bit machine, this creates a usize by concatenating the hi and lo 32-bit
+/// values.
+///
+/// TODO
+/// ----
+///
+/// This can be more succinctly implemented using
+/// [`unbounded_shl()`](https://doc.rust-lang.org/stable/std/primitive.usize.html#method.unbounded_shl).
+/// However, that method is currently a nightly-only feature.
+#[inline]
+pub const fn usize32_to_usize(lo: usize, hi: usize) -> usize {
```

Comment:
Why is this not accepting `u32`s?

```suggestion
pub const fn usize32_to_usize(lo: u32, hi: u32) -> usize {
```

Reply:
This is exactly what confused me as well. I _think_ Brad's intent here is to be able to go "directly" from syscall interface types to this method, portably across a 32- or 64-bit implementation?

Reply:
Because our system call command interface uses `usize`, so that is the conversion we need currently.